pkg/cpio/newc/newc_test.go

```diff
@@ -480,3 +482,73 @@ var (
 		"usr/lib: Ino 3204042 Mode 040755 UID 0 GID 0 NLink 2 MTime 2017-01-23 06:18:50 +0000 UTC FileSize 0 Major 252 Minor 0 Rmajor 0 Rminor 0",
 	}
 )
+
+// TestReproducible verifies that we can produce reproducible cpio archives for newc format.
+func TestReproducible(t *testing.T) {
+	f, err := cpio.Format("newc")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	contents := "LANAAAAAAAAAA"
+	rec := cpio.StaticRecord(contents, cpio.Info{
+		Ino:      1,
+		Mode:     syscall.S_IFREG | 2,
+		UID:      3,
+		GID:      4,
+		NLink:    5,
+		MTime:    6,
+		FileSize: 7,
+		Major:    8,
+		Minor:    9,
+		Rmajor:   10,
+		Rminor:   11,
+		Name:     "foobar",
+	})
+
+	// First test that it fails unless we make it reproducible
+
+	b1 := &bytes.Buffer{}
+	w := f.Writer(b1)
+	if err := w.WriteRecord(rec); err != nil {
+		t.Errorf("Could not write record %q: %v", rec.Name, err)
+	}
+
+	b2 := &bytes.Buffer{}
+	w = f.Writer(b2)
+	if err := w.WriteRecord(rec); err != nil {
+		t.Errorf("Could not write record %q: %v", rec.Name, err)
+	}
+
+	if reflect.DeepEqual(b1.Bytes()[:], b2.Bytes()[:]) {
```

Comment:
it can't contain a readerat because it's possible to have a file name which can't implement seek. E.g. on plan 9 for example it is legal to cpio up a directory containing a bunch of sockets. This is kinda weird but you can cpio up those sockets into an archive and write them out as files ... We have to be a bit careful about Unix assumptions, they're easy to bake into code and they're usually wrong. I'll clean up this test.